src/Compiler/Checking/PostInferenceChecks.fs

```diff
@@ -347,6 +349,18 @@ type TypeInstCtx =
                         match x with
                         | IlGenericInst(_,ilTypar) -> ilTypar.HasAllowsRefStruct
                         | _ -> false
+      
+/// Check the attributes of a measure
+[<return:Struct>]
+let rec (|MeasureAttrib|_|) measure =
+    match measure with
+    | Measure.Const(tcref, _) -> ValueSome(tcref.Attribs)
+    | Measure.Prod(measure1= MeasureAttrib ms1; measure2= MeasureAttrib ms2) -> ValueSome (ms1 @ ms2)
+    | Measure.Inv(MeasureAttrib ms) -> ValueSome ms
+    | Measure.One(m) -> ValueNone
+    | Measure.RationalPower(measure = MeasureAttrib ms) -> ValueSome ms
+    | Measure.Var _ -> ValueNone
+    | _ -> ValueNone
 
 let rec CheckTypeDeep (cenv: cenv) (visitTy, visitTyconRefOpt, visitAppTyOpt, visitTraitSolutionOpt, visitTyparOpt as f) (g: TcGlobals) env (typeInstParentOpt:TypeInstCtx) ty =
```

Comment:
Can you please follow all other use cases for CheckTypeDeep beyond ObsoleteChecks, and make sure no regression is possible via a few tests?
i.e. that an orthogonal check, which also uses CheckTypeDeep, will not start producing new unexpected warnings.